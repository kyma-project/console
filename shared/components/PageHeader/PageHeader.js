import React from 'react';
import PropTypes from 'prop-types';
import { Panel } from 'fundamental-react';
import './PageHeader.scss';

export const PageHeader = ({ title }) => (
  <Panel className="page-header">
    <Panel.Header className="fd-has-padding-m">
      <Panel.Head title={title} />
    </Panel.Header>
  </Panel>
);

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
};
