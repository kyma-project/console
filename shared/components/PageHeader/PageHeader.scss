.page-header {
  .fd-panel__title {
    font-size: 24px;
  }
}

.header-wrapper {
  flex-grow: 1;
}
.column-wrapper {
  display: grid;
  grid-gap: 8px;
  margin-top: 8px;

  grid-template-columns: repeat(auto-fill, minmax(20em, 1fr));
}
.page-header__column {
  display: inline-block;
  .title,
  .content {
    display: block;
  }

  .title {
    color: var(--fd-color-text-4);
  }
}
