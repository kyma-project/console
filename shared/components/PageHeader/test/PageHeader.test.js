import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PageHeader } from '../PageHeader';

describe('PageHeader', () => {
  it('Renders with minimal props', () => {
    const component = shallow(<PageHeader title="page title" />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
