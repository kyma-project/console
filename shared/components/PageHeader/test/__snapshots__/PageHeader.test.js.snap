// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Renders with minimal props 1`] = `
<Panel
  className="page-header"
>
  <Panel.Header
    className="fd-has-padding-m"
  >
    <Panel.Head
      headingLevel={3}
      title="page title"
    />
  </Panel.Header>
</Panel>
`;
